src/lightning/pytorch/plugins/precision/amp.py

```diff
@@ -36,7 +36,7 @@ class MixedPrecisionPlugin(PrecisionPlugin):
     def __init__(
         self, precision: Literal["16", 16, "bf16"], device: str, scaler: Optional[torch.cuda.amp.GradScaler] = None
     ) -> None:
-        self.precision = cast(Literal["16", "bf16"], str(precision))
+        self.precision = cast(Literal["16", "bf16"], str(precision))  # type: ignore
```

Comment:
**NOTE:** All the `type: ignore` changes are only done here to satisfy mypy, until the PL codebase was updated to the new precision types as well. They will be removed in this follow-up PR.